pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -852,19 +845,38 @@ func buildHTTPDestination(
 	if len(weights) == 1 {
 		weights = []int{0}
 	}
+
+	var invalidBackendErr *ConfigError
+	refs := ctx.AllowedReferences
 	res := []*istio.HTTPRouteDestination{}
 	for i, fwd := range action {
+		invalid := false
 		if toNs := fwd.BackendRef.Namespace; toNs != nil && string(*toNs) != ns {
 			if !refs.BackendAllowed(gvk.HTTPRoute, fwd.BackendRef.Name, *toNs, ns) {
-				return nil, &ConfigError{
+				invalidBackendErr = &ConfigError{
 					Reason:  InvalidDestinationPermit,
 					Message: fmt.Sprintf("backendRef %v/%v not accessible to a route in namespace %q (missing a ReferenceGrant?)", fwd.BackendRef.Name, *toNs, ns),
 				}
+				invalid = true
+				// keep going, we will gracefully drop invalid backends
 			}
 		}
-		dst, err := buildDestination(fwd.BackendRef, ns, domain)
+		dst, err := buildDestination(ctx, fwd.BackendRef, ns)
 		if err != nil {
-			return nil, err
+			if isInvalidBackend(err) {
+				// keep going, we will gracefully drop invalid backends
+				invalidBackendErr = err
+				invalid = true
+			} else {
+				return nil, err
+			}
+		}
+		if invalid {
+			// this backend is invalid, set a nonexistent host so that envoy will return 500 when traffic goes to this cluster
+			dst = &istio.Destination{
+				Host: "zzz-invalid-backend.istio.io",
```

Comment:
I don't feel very comfortable with this string and random port. This will get exposed indirectly to user through configuration.

Maybe we can do this:
* Set `dst = istio.Destination{}`
* Make validation allow empty destination IF `internal.istio.io/route-semantics` is set
* In route generation, if we have empty destination replace it with a cluster `UnknownService` (which doesn't exist)